snippets/stripe-webhook.md

```diff
@@ -0,0 +1,159 @@
+# Stripe Webhook Integration  
+
+This introduces how to create an API with Hono to receive Stripe Webhook events.
+
+## Preparation
+
+Please install the official Stripe SDK at first:
+
+```bash
+npm install stripe
+```
+
+And put the following values on the `.dev.vars` file to insert the Stripe API keys:
+
+```
+STRIPE_API_KEY=sk_test_xxx
+STRIPE_WEBHOOK_SECRET=whsec_xxx
+```
+
+You can learn about the Stripe API keys by the following documents:
+
+- Secret Key: https://docs.stripe.com/keys
+- Webhook secret: https://docs.stripe.com/webhooks
+
+
+## How to protect the API for Stripe Webhook events
+
+The API that processes webhook events is publicly accessible.
+
+```javascript
+import Stripe from 'stripe';
+import { Hono } from 'hono';
+const app = new Hono();
+
+app.post("/webhook", async (context) => {
+    const stripe = new Stripe(context.env.STRIPE_API_KEY);
+    const event = await context.req.json();
+    switch(event.type) {
+        case "payment_intent.created": {
+            console.log(event.data.object)
+            break
+        }
+        default:
+            break
+    }
+    return context.text("", 200);
+})
+
+export default app;
+```
+
+Therefore, a mechanism is needed to protect it from attacks such as malicious third parties spoofing Stripe's webhook event objects and sending requests. In Stripe's case, you can protect the API by issuing a webhook secret and verifying each request.
+
+Learn more: https://docs.stripe.com/webhooks?lang=node#verify-official-libraries  
+
+## Implementing the Webhook API by hosting environment or framework
+To perform signature verification with Stripe, the raw request body is needed.
+When using a framework, you need to ensure that the original body is not modified. If any changes are made to the raw request body, the verification will fail.
+
+Here, we introduce implementation methods for major hosting environments and frameworks.
+
+### Deploying to Cloudflare ( Workers / Pages )
+
+When processing Stripe webhook events on Cloudflare Workers or Cloudflare Pages functions, the raw request body can be obtained from `context.req.text()`.
```

Comment:
I think both Cloudflare and Node.js can use the same code because the API for accessing the request body is the same according to Web Standard APIs. So, both can use `context.req.text()` or `context.req.arrayBuffer()`.